qa/L0_client_memory_growth/test.sh

```diff
@@ -57,55 +58,62 @@ cp libidentity.so $DATADIR/custom_identity_int32/1/.
 
 RET=0
 
-# Threshold memory growth in MB
-MAX_ALLOWED_ALLOC="50"
-export MAX_ALLOWED_ALLOC
-
 # Run test for both HTTP and GRPC, not re-using client object. 
-# 100000 inferences in each case.
-EXTRA_ARGS="-r 100000"
 for PROTOCOL in http grpc; do
-    LEAKCHECK_LOG="./valgrind.${PROTOCOL}.c++.log"
-    CLIENT_LOG="./client.${PROTOCOL}.c++.log"
-    MASSIF_LOG="./${PROTOCOL}.c++.massif"
-    LEAKCHECK_ARGS="$LEAKCHECK_ARGS_BASE --log-file=$LEAKCHECK_LOG --massif-out-file=$MASSIF_LOG"
-    EXTRA_CLIENT_ARGS="${EXTRA_ARGS} -i ${PROTOCOL}"
-
-    run_server
-    if [ "$SERVER_PID" == "0" ]; then
-        echo -e "\n***\n*** Failed to start $SERVER\n***"
-        cat $SERVER_LOG
-        exit 1
-    fi
-
-    $LEAKCHECK $LEAKCHECK_ARGS $MEMORY_GROWTH_TEST $EXTRA_CLIENT_ARGS >> ${CLIENT_LOG} 2>&1
-    if [ $? -ne 0 ]; then
-        cat ${CLIENT_LOG}
-        RET=1
-        echo -e "\n***\n*** Test FAILED\n***"
-    else
-        check_valgrind_log $LEAKCHECK_LOG
-        if [ $? -ne 0 ]; then
-        echo -e "\n***\n*** Memory leak detected\n***"
-        RET=1
+    for LANG in c++ python; do
+        LEAKCHECK_LOG="./valgrind.${PROTOCOL}.${LANG}.log"
+        CLIENT_LOG="./client.${PROTOCOL}.${LANG}.log"
+        MASSIF_LOG="./${PROTOCOL}.${LANG}.massif"
+        LEAKCHECK_ARGS="$LEAKCHECK_ARGS_BASE --log-file=$LEAKCHECK_LOG --massif-out-file=$MASSIF_LOG"
+
+        run_server
+        if [ "$SERVER_PID" == "0" ]; then
+            echo -e "\n***\n*** Failed to start $SERVER\n***"
+            cat $SERVER_LOG
+            exit 1
+        fi
+
+        # MAX_ALLOWED_ALLOC is the threshold memory growth in MB
+        if [ "$LANG" == "c++" ]; then
+            MEMORY_GROWTH_TEST=$MEMORY_GROWTH_TEST_CPP
+            MAX_ALLOWED_ALLOC="10"
+            EXTRA_ARGS="-r 100000 -i ${PROTOCOL}"
+        else
+            MEMORY_GROWTH_TEST="python $MEMORY_GROWTH_TEST_PY"
+            MAX_ALLOWED_ALLOC="1"
+            EXTRA_ARGS="-r 10000 -i ${PROTOCOL}"
```

Comment:
Why Python would have smaller reps?

Reply:
The python client is significantly slower for some reason. This is in the case of GRPC.